### ISSUE TITLE
[Bug]: [BETA] [ERR_SOCKET_DGRAM_NOT_RUNNING] Error that makes HB restart every time stream stops

### ISSUE THREAD

**Issue by njuart** [samemory/homebridge-eufy-security#321]
### What happened?

Hi, while testing BETA I experience this error [ERR_SOCKET_DGRAM_NOT_RUNNING]. It makes HB restart every time I stop stream within Home app. As I've put Eufy plugin in child bridge, it restart only child bridge, so it helps a bit. As I recall, this issue is present since earliest BETA and was also reported by other users. Can you please have a look? Thx in advance for this amazing plugin!

### Device Type

Indoor Camera (Supported)

### Plugin Version

beta (Best Effort)

### HomeBridge Version

v1.3.x (Supported)

### NodeJS Version

v16 (Supported)

### Operating System

Raspbian 10+ (Supported)

### Relevant log output

```shell
[03/01/2022, 18:35:56] [homebridge-eufy-security] Eufy camera Starting video stream: 1280 x 720, 30 fps, 299 kbps (AAC-eld)
[03/01/2022, 18:36:05] [homebridge-eufy-security] Eufy camera Eufy Station stopped the stream. Stopping stream.
[03/01/2022, 18:36:05] [homebridge-eufy-security] Eufy camera Stopped video stream.
[03/01/2022, 18:36:05] [homebridge-eufy-security] Eufy camera Error occurred closing socket: Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
[03/01/2022, 18:36:05] [homebridge-eufy-security] Eufy camera Stopped video stream.
Error: read ECONNRESET
    at Pipe.onStreamRead (node:internal/stream_base_commons:220:20)
[03/01/2022, 18:36:06] [homebridge-eufy-security] Child bridge process ended
[03/01/2022, 18:36:06] [homebridge-eufy-security] Process Ended. Code: 1, Signal: null
[03/01/2022, 18:36:13] [homebridge-eufy-security] Restarting Process...
```

**Comment by lenoxys** [comment]
can you enable debug ? while doing the test ?
https://github.com/samemory/homebridge-eufy-security/wiki/Basic-Troubleshooting#debug-mode

**Comment by njuart** [comment]
> 

Done, it's log since stream starts and until I stop it

[homebridge.log.txt](https://github.com/samemory/homebridge-eufy-security/files/7803287/homebridge.log.txt)

**Comment by njuart** [comment]
> can you enable debug ? while doing the test ?
> https://github.com/samemory/homebridge-eufy-security/wiki/Basic-Troubleshooting#debug-mode

Done, it's log since stream starts and until I stop it

[homebridge.log.txt](https://github.com/samemory/homebridge-eufy-security/files/7803287/homebridge.log.txt)

**Comment by schliemann** [comment]
I'm experiencing the same. Tried to tinker around with it myself without succeeding.

The refactoring you have made, @lenoxys, is however very nice.

**Comment by schliemann** [comment]
Got another idea. Will test tonight.

**Comment by lenoxys** [comment]
the timeout is hardcoded within @bropat lib
https://github.com/bropat/eufy-security-client/blob/master/src/eufysecurity.ts#L38

It could be nice if we can change that.

**Comment by bropat** [comment]
@lenoxys 

That is wrong. This is the default value.

See [here](https://github.com/bropat/eufy-security-client/blob/ff14851faaeb00055da6df2a3625c9fc9840458d/src/eufysecurity.ts#L500-L506)

**Comment by njuart** [comment]
Will gladly help to test if needed

**Comment by lenoxys** [comment]
Sorry miss the function !! thanks for your feedback !!

**Comment by schliemann** [comment]
@lenoxys I think I know what our problem is. We need to destroy the readable stream in the UniversalStream.ts.

```
import net from 'net'
import fs from 'fs'
import os from 'os'
import { Readable } from 'stream';

let counter = 0

export class NamePipeStream {
    private server;
    private sock_id;
    private log;
    public url;
    private stream: Readable;

    constructor(stream, onSocket, log) {
        this.log = log;
        this.stream = stream;
        let path

        this.sock_id = ++counter

        const osType = os.type()
        if (osType === 'Windows_NT') {
            path = '\\\\.\\pipe\\stream' + (++counter)
            this.url = path
        } else {
            path = './' + (++counter) + '.sock'
            this.url = 'unix:' + path
        }

        try {
            fs.statSync(path)
            fs.unlinkSync(path)
        } catch (err) { }
        this.server = net.createServer(onSocket)
        stream.on('finish', () => {
            this.server.close()
        })

         this.server.on('error', (err: Error) => {
             this.log.error('Error in NamePipe', err.message, err.stack);
         });

         this.server.on('close', (err: Error) => {
            this.log.error('Error in NamePipe', err?.message, err?.stack);
        });


        this.server.listen(path)
    }
    close() {
        this.log.info('Trying to close NamePipeStream.')
        if (this.server)
        {
            this.stream.destroy();
            
            this.server.close();
            this.log.info('Closed NamePipeStream')
        }
    }
}

export function StreamInput(stream, log) {
    return new NamePipeStream(stream, socket => stream.pipe(socket), log)
}

export function StreamOutput(stream, log) {
    return new NamePipeStream(stream, socket => socket.pipe(stream), log)
}

```

**Comment by schliemann** [comment]
When doing this the audio drags on for some time. I think the rtcp timeout is longer than the video?

But lets me exit ffmpeg without crashing.

![image](https://user-images.githubusercontent.com/10818044/148129634-4a01b7ea-6b47-4523-9775-9aecba366308.png)

**Comment by schliemann** [comment]
Also I think we should close the uaudiostream and uvideostream before we call the stopStream function. Otherwise the Readable is not terminated soon enough.

                    uVideoStream.close();
                    uAudioStream.close();
                    this.stopStream(request.sessionID);

**Comment by schliemann** [comment]
To fix the stream you have to add unpipe the readable stream in UniversalStream.ts.

this.stream.unpipe();

**Comment by schliemann** [comment]
Also we should add a null guard on:

                     this.server.on('error', (err: Error) => {
                         this.log.error('Error in NamePipe', err.message, err.stack);
                     });

to this:

                     this.server.on('error', (err: Error) => {
                         this.log.error('Error in NamePipe', err?.message, err?.stack);
                     });

**Comment by schliemann** [comment]
@lenoxys Do you want me to make the change or would you like to test yourself?

**Comment by lenoxys** [comment]
Let me test

**Comment by lenoxys** [comment]
yes indeed it's better that way !

**Comment by schliemann** [comment]
Also it seems like the timeout for the audio is a bit long. Should the stream be stopped when the user leaves the stream in homekit? Here:

https://github.com/samemory/homebridge-eufy-security/blob/7584d96b4a9a21baa260f8ec96342d5d7e83d939/src/accessories/streamingDelegate.ts#L498

Because as it is we actually don't end the uVideoStream and uAudioStream there. So the streams continues until the timeout it seems.

**Comment by schliemann** [comment]
We could of course add both uAudioStream and uVideoStream to the activeSession and just call stopStream directly every time. Just can't seem to make it work.

**Comment by schliemann** [comment]
Or we should find another way to gracefully stop the namePibeStream instead. Because we can never ensure that the stream won't end before we close our pipe.

If the station stops the stream this error will always bubble through and kill our process.

**Comment by lenoxys** [comment]
> We could of course add both uAudioStream and uVideoStream to the activeSession and just call stopStream directly every time. Just can't seem to make it work.

Would be proper approach at least if we rely on other thing

**Comment by schliemann** [comment]
@lenoxys If you change the pipe here:

https://github.com/samemory/homebridge-eufy-security/blob/e38d24a8ffd1e77dcc63fff9712910f7ff3c209f/src/accessories/UniversalStream.ts#L63

to this:
             return new NamePipeStream(stream, socket => stream.pipe(socket).on('error', function(e){log.error(e)}), who, log)

We no longer break hard on econnreset. This is probably a good idea since it is not in our control.

**Comment by schliemann** [comment]
Maybe the pipe creation should also be moved into the class instead of through the constructor. And then on error should call close?

**Comment by schliemann** [comment]
I have some changes I will test later tonight regarding the UniversalStream.

**Comment by lenoxys** [comment]
> Maybe the pipe creation should also be moved into the class instead of through the constructor. And then on error should call close?

agree this need to be rewrite with cleaner manner. I was using this part from @bropat script :

https://github.com/bropat/ioBroker.eufy-security/blob/master/src/lib/video.ts#L14-L65

But we should avoid passing a function into argument of a function.

**Comment by lenoxys** [comment]
> I have some changes I will test later tonight regarding the UniversalStream.

I let you follow this part I will be off couple of week. thanks for your help on this !

**Comment by schliemann** [comment]
@lenoxys Can you test the changes to the beta streaming before we make a new version with the workflow?

**Comment by schliemann** [comment]
@njuart Would you be able to test beta v1.1.1-beta.14?

**Comment by njuart** [comment]
> @njuart Would you be able to test beta v1.1.1-beta.14?

Sure, with pleasure 👍

**Comment by njuart** [comment]
> @njuart Would you be able to test beta v1.1.1-beta.14?

Just checked latest beta.14, great news! HB doesn't crash and restart anymore after closing stream. Video is a bit laggy, sound is present. I also see this in logs:

> [07/01/2022, 10:32:14] [homebridge-eufy-security] Eufy camera Starting video stream: 1280 x 720, 30 fps, 299 kbps (AAC-eld)
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Stopping session with id: 0ddec022-cec3-44a9-afc8-36019e907688
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Eufy camera Eufy Station stopped the stream. Stopping stream.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Stopping session with id: 131b8a1c-10b5-4887-989d-4f977128f4ab
> [07/01/2022, 10:32:23] [homebridge-eufy-security] No session to stop.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Eufy camera Eufy Station stopped the stream. Stopping stream.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Stopping session with id: e06ddeda-0995-4bfc-8dab-b124cd9a067a
> [07/01/2022, 10:32:23] [homebridge-eufy-security] No session to stop.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Eufy camera Eufy Station stopped the stream. Stopping stream.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Stopping session with id: d210ffa7-4d67-4781-bbd4-cbfa4b638b45
> [07/01/2022, 10:32:23] [homebridge-eufy-security] No session to stop.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Eufy camera Eufy Station stopped the stream. Stopping stream.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Stopping session with id: 0ddec022-cec3-44a9-afc8-36019e907688
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Eufy camera Stopped video stream.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Eufy camera Stopped video stream.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Eufy camera Error occurred closing socket: Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
> [07/01/2022, 10:32:23] [homebridge-eufy-security] Connection closed by Eufy station.
> [07/01/2022, 10:32:23] [homebridge-eufy-security] ErrorCode ECONNRESET

**Comment by schliemann** [comment]
Perfect. There is obviously room for some logging cleanup :). But I will not prioritize this at the moment.

**Comment by njuart** [comment]
> Perfect. There is obviously room for some logging cleanup :). But I will not prioritize this at the moment.

Thx for the efforts! 🤝 Since we're discussing stream here, is there any way I can increase streaming quality (or play with it to adjust)? Currently It looks pixelated a bit (comparing to regular web stream).

**Comment by schliemann** [comment]
@njuart Difficult for me to say. It might be the way we use ffmpeg. I would think that the stream from the client library is not the issue - since there is no issue on his github regarding the stream quality.

**Comment by njuart** [comment]
@schliemann Also noticed that stream is live for about 10-15 seconds, then image freezes and after another 10-15 seconds it shows No response.

**Comment by schliemann** [comment]
> @schliemann Also noticed that stream is live for about 10-15 seconds, then image freezes and after another 10-15 seconds it shows No response.

Interesting. Mine will run for the maximum periode of 30 seconds and the the station kills it as expected.


https://user-images.githubusercontent.com/10818044/148523296-8dd1af4f-1263-49ca-878f-13aae63e8cfb.MOV

**Comment by fullphat** [comment]
I don't know if it helps (I have the floodlight cam), but I've just swapped to the latest beta and I had to manually change the video codec in streamingDelegate.js to "copy".  After doing this, local streaming works fine for me.

**Comment by schliemann** [comment]
@fullphat Can you try and add a logger line and print the this.videoConfig.vcodec you have when trying to start a stream. I am wondering why you don't have copy already.

**Comment by fullphat** [comment]
Done!

I get "undefined" in the log output.

**Comment by schliemann** [comment]
Ok. I'll add a null check then. But would you mind making an issue with bropat, as well?

**Comment by fullphat** [comment]
Yes, of course.  What do I need to tell him?

**Comment by schliemann** [comment]
> Yes, of course. What do I need to tell him?

Wait. It might be our plugin. Let me have a look.

**Comment by schliemann** [comment]
> > Yes, of course. What do I need to tell him?
> 
> Wait. It might be our plugin. Let me have a look.

@fullphat It's our bug. For some reason the floodlight doesn't like codec libx264. I'll put in copy later tonight.

**Comment by schliemann** [comment]
@fullphat Can't get my doorbell to work with 'copy'. Can you share your code-snippet?

**Comment by schliemann** [comment]
Ah. Problem is probably that the codec for the doorbell is not H.264 based.

_Set the codec used for encoding video sent to HomeKit, must be H.264-based. You can change to a hardware accelerated video codec with this option, if one is available. (Default: libx264)_

**Comment by schliemann** [comment]
@fullphat 

Can you try and replace the following in your code and see if streaming works:

           const vcodec  = this.videoConfig.vcodec || 'libx264';

with this:

            let vcodec = '';
            switch(streamData.metadata.videoCodec) {
                case VideoCodec.H264:
                    vcodec = 'copy';
                    break;
                case VideoCodec.H265:
                case VideoCodec.UNKNOWN:
                    vcodec = 'libx264';
                    break;
            }

**Comment by fullphat** [comment]
My code snippet is:

```
            // const vcodec = this.videoConfig.vcodec || 'libx264';
            const vcodec = 'copy';
            this.log.debug('vcodec:', this.videoConfig.vcodec);
```

But I appreciate this will probably break everything else!  😄 

I tried your code, but I get this in the log:

```
[1/8/2022, 9:16:30 AM] [EufySecurity-1.1.1-beta.14] DEBUG: videoConfig->vcodec: undefined
[1/8/2022, 9:16:30 AM] [EufySecurity-1.1.1-beta.14] DEBUG: streamData->metadata->videoCodec: -1
(node:59117) UnhandledPromiseRejectionWarning: ReferenceError: VideoCodec is not defined
    at StreamingDelegate.startStream (/usr/local/lib/node_modules/homebridge-eufy-security/src/accessories/streamingDelegate.ts:400:20)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
```

Where should the `VideoCodec` enum be defined?  I don't seem to have that...

**Comment by schliemann** [comment]
@fullphat Ah. Probably forgot an import to give you as well. Will find it.

**Comment by schliemann** [comment]
import { Station, Camera, PropertyName, StreamMetadata, VideoCodec } from 'eufy-security-client';

**Comment by fullphat** [comment]
Where do I paste this line?  I tried putting it at the top of streamingDelegate.js, but I just get a syntax error.

**Comment by schliemann** [comment]
https://github.com/samemory/homebridge-eufy-security/blob/7adf974ec5fd1f7840b6055110edbf2ad5c6586e/src/accessories/streamingDelegate.ts#L30

**Comment by schliemann** [comment]
Odd. It should already be there. Can you confirm you have it?

**Comment by fullphat** [comment]
This is what I have - note I'm using the Beta.14 plugin:

```
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingDelegate = void 0;
const child_process_1 = require("child_process");
const dgram_1 = require("dgram");
const ffmpeg_for_homebridge_1 = __importDefault(require("ffmpeg-for-homebridge"));
const pick_port_1 = __importDefault(require("pick-port"));
const ffmpeg_1 = require("./ffmpeg");
const eufy_security_client_1 = require("eufy-security-client");
const UniversalStream_1 = require("./UniversalStream");
const fs_1 = require("fs");
const util_1 = require("util");
const readFileAsync = (0, util_1.promisify)(fs_1.readFile), SnapshotUnavailablePath = require.resolve('../../media/Snapshot-Unavailable.png');
class StreamingDelegate {
    constructor(platform, device, cameraConfig, api, hap) {
        // keep track of sessions
        this.pendingSessions = new Map();
        this.ongoingSessions = new Map();
        this.timeouts = new Map();
        this.log = platform.log;
        this.hap = hap;
        this.platform = platform;
        this.device = device;
        this.cameraName = device.getName();
        this.unbridge = false;
        this.videoConfig = cameraConfig.videoConfig;
        this.videoProcessor = ffmpeg_for_homebridge_1.default || 'ffmpeg';
        api.on("shutdown" /* SHUTDOWN */, () => {
            for (const session in this.ongoingSessions) {
                this.stopStream(session);
            }
        });
```

**Comment by schliemann** [comment]
Well. The codec resides in this require. I'm not an expert on how require importing works.

const eufy_security_client_1 = require("eufy-security-client");

**Comment by fullphat** [comment]
Got it working!

Had to reference `eufy_security_client_1` first and also had to change the logic slightly as the T8420 seems to be generating an unknown stream (`eufy_security_client_1.VideoCodec.UNKNOWN`).  Here's the code that works for me:

```
this.log.info('videoConfig->vcodec:', this.videoConfig.vcodec);
            this.log.info('streamData->metadata->videoCodec:', streamData.metadata.videoCodec);

            let vcodec = '';
            switch(streamData.metadata.videoCodec) {
              // case -1:
              //   vcodec = 'copy';
              //   break;

              case eufy_security_client_1.VideoCodec.H264:
              case eufy_security_client_1.VideoCodec.UNKNOWN:
                  vcodec = 'copy';
                  break;
              case eufy_security_client_1.VideoCodec.H265:
                  vcodec = 'libx264';
                  break;
            }

            this.log.info('vcodec:', vcodec);
```

And here's the resulting output in the log:

```
[1/8/2022, 10:02:25 AM] [EufySecurity-1.1.1-beta.14] DEBUG: ReqEufy: {"videoCodec":-1,"videoFPS":15,"videoHeight":1080,"videoWidth":1920,"audioCodec":1}
[1/8/2022, 10:02:25 AM] [EufySecurity-1.1.1-beta.14] INFO:  videoConfig->vcodec: undefined
[1/8/2022, 10:02:25 AM] [EufySecurity-1.1.1-beta.14] INFO:  streamData->metadata->videoCodec: -1
[1/8/2022, 10:02:25 AM] [EufySecurity-1.1.1-beta.14] INFO:  vcodec: copy
[1/8/2022, 10:02:25 AM] [EufySecurity-1.1.1-beta.14] DEBUG: Floodlight Cam Video stream requested: 1280 x 720, 30 fps, 299 kbps true
[1/8/2022, 10:02:25 AM] [EufySecurity-1.1.1-beta.14] INFO:  Floodlight Cam Starting video stream: native x native, native fps, ??? kbps (AAC-eld)
[1/8/2022, 10:02:25 AM] [EufySecurity-1.1.1-beta.14] DEBUG: Floodlight Cam_video Stream command: /usr/local/lib/node_modules/homebridge-eufy-security/node_modules/ffmpeg-for-homebridge/ffmpeg -use_wallclock_as_timestamps 1 -r 15 -i unix:./6.sock -an -sn -dn -codec:v copy -pix_fmt yuv420p -color_range mpeg -payload_type 99 -ssrc 10499619 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params wQbyHGSWG7gWNwXZTiKNrjuGHvOy7c7DTE+41av6 srtp://192.168.1.114:49633?rtcpport=49633&pkt_size=1316 -loglevel level+verbose -progress pipe:1
```

**Comment by schliemann** [comment]
Hm. What happens with your stream when you have changed nothing and it uses libx264.

Basically the idea of my code change was to change everything not libx264 to 264. Since this is the only format homekit accepts.

**Comment by schliemann** [comment]
In theory a copy on unknown could cause an error if the incomming encoding is actually not libx264. Can we ask bropat if something is wrong with the client library?

**Comment by fullphat** [comment]
> Hm. What happens with your stream when you have changed nothing and it uses libx264.
> 
> Basically the idea of my code change was to change everything not libx264 to 264. Since this is the only format homekit accepts.

So if I don't use `copy` as the codec, I don't get any video in the Home app.  In the HomeBridge log, it looks like video is coming through as I can see the frames, but the Home app just sits spinning.  

As soon as a I make ffmpeg use `copy` the stream springs to life immediately (it does sometimes crash after a few seconds but restarting it works).

**Comment by fullphat** [comment]
> In theory a copy on unknown could cause an error if the incoming encoding is actually not libx264. Can we ask bropat if something is wrong with the client library?

Done!

https://github.com/bropat/eufy-security-client/issues/97

**Comment by schliemann** [comment]
Got snapshot working. Still slow and the colors look weird however.


https://user-images.githubusercontent.com/10818044/148776194-e4eaa6f8-79b8-4495-b826-255e326008ae.MOV

**Comment by fullphat** [comment]
Is this available to test? 😊

**Comment by schliemann** [comment]
Not yet. And not sure this implementation is a good solution. The snapshot still takes far too long. Will try and determine the bottleck. But the problem is that while generating the snapshot it blocks the rest of the plugin.

**Comment by fullphat** [comment]
Ok - good that it's do-able though 👍🏼

**Comment by lenoxys** [comment]
> @fullphat Not yet. And not sure this implementation is a good solution. The snapshot still takes far too long. Will try and determine the bottleck. But the problem is that while generating the snapshot it blocks the rest of the plugin.

Looks like apple iOS issue

https://piunikaweb.com/2022/01/03/apple-homekit-app-camera-thumbnails-or-snapshots-not-updating/

**Comment by fullphat** [comment]
> > @fullphat Not yet. And not sure this implementation is a good solution. The snapshot still takes far too long. Will try and determine the bottleck. But the problem is that while generating the snapshot it blocks the rest of the plugin.
> 
> Looks like apple iOS issue
> 
> https://piunikaweb.com/2022/01/03/apple-homekit-app-camera-thumbnails-or-snapshots-not-updating/

I'm not sure - I have seen this a few times, but since upgrading to 15.2 everything has been generally ok.  I run one of my Eufy P&T 2K cameras through the homebridge-ffmpeg plugin and that updates it's thumbnail fine too.

**Comment by schliemann** [comment]
My thumbnail was stuck yesterday. When clearing cache it worked again.

**Comment by schliemann** [comment]
But what I was referring to is that generating the snapshot from the readable stream takes 5-6 seconds.

**Comment by schliemann** [comment]
Ok. So. Got some progress made today. Now the code can generate snapshots a bit faster. Also - I finally figured out why night mode wasn't working on the doorbell. Turns out that the first frame of the stream is not with night vision. So added some frame skipping.


Uploading IMG_4929.MOV…

**Comment by schliemann** [comment]
@fullphat Would you be willing to try and test the new snapshot at some point before it is added to the beta?

**Comment by fullphat** [comment]
> @fullphat Would you be willing to try and test the new snapshot at some point before it is added to the beta?

Yes absolutely! 😊

**Comment by schliemann** [comment]
I'll clean up the code a bit first and then i'll send it in this thread.

**Comment by schliemann** [comment]
@fullphat Do you want it in ts or js?

**Comment by fullphat** [comment]
JS if possible as I can then just drop it straight in. 👍🏼

**Comment by schliemann** [comment]
[streamingDelegate.log](https://github.com/samemory/homebridge-eufy-security/files/7953990/streamingDelegate.log)

**Comment by schliemann** [comment]
@fullphat Couldn't upload js files. Please rename extention to .js. Then you should be set.

**Comment by fullphat** [comment]
Working a treat!  👍  Not seeing any lag, and getting a new snapshot every 10 seconds in the Home app.

**Comment by schliemann** [comment]
Great. I'll clean up the code in the weekend then.

**Comment by schliemann** [comment]
Maybe we can also clean up the way we local stream video. It might solve the issue with not being able to save files to current directory. I'll take a look at that next.

**Comment by fullphat** [comment]
Sounds good.  I have spotted an issue - the camera has gone "no response" and checking the logs after restarting both the specific bridge and all of HB, I get this:

![image](https://user-images.githubusercontent.com/11242840/151521093-b2c9de65-43b3-4cbe-b684-4497d760060c.jpeg)

Sorry it's an image - using my phone.  Seems the plugin is crashing when trying to open a socket?

**Comment by schliemann** [comment]
@fullphat Was it caused by livestream or snapshot?

**Comment by fullphat** [comment]
I'm not sure - I opened the Home app and the camera was showing as no response.

**Comment by schliemann** [comment]
Try running sudo lsof -i :41089

This should show you what is using this address.

**Comment by njuart** [comment]
Hi @schliemann I've also checked flow with updated streamingDelegate you've provided. Works flawlessly with my Indoor 2k . Snapshots are being updated every 10 seconds, also starting / ending stream works fine. Meanwhile in logs I see this, every time snapshot is updated, but I suggest it's expected currently

> [28/01/2022, 12:53:50] [homebridge-eufy-security] handleSnapshotRequest
[28/01/2022, 12:53:50] [homebridge-eufy-security] snapshotPromise: false
[28/01/2022, 12:53:50] [homebridge-eufy-security] Calling fetchSnapshot
[28/01/2022, 12:53:50] [homebridge-eufy-security] isLiveStreaming: false
[28/01/2022, 12:53:51] [homebridge-eufy-security] ffmpeg version 4.3.1 Copyright (c) 2000-2020 the FFmpeg developers Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   built with gcc 6.3.0 (Raspbian 6.3.0-18+rpi1+deb9u1) 20170516 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   configuration: --pkgconfigdir=/build/workspace/lib/pkgconfig --prefix=/build/workspace --pkg-config-flags=--static --extra-cflags=-I/build/workspace/include --extra-ldflags=-L/build/workspace/lib --extra-libs='-lpthread -lm' --enable-static --disable-debug --disable-shared --disable-ffplay --disable-doc --enable-openssl --enable-gpl --enable-version3 --enable-nonfree --enable-pthreads --enable-libvpx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libx264 --enable-runtime-cpudetect --enable-libfdk-aac --enable-avfilter --enable-libopencore_amrwb --enable-libopencore_amrnb --enable-filters --enable-decoder=h264 --enable-network --enable-protocol=tcp --enable-libspeex --enable-demuxer=rtsp --enable-omx-rpi --enable-mmal --enable-indev=alsa --enable-outdev=alsa Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libavutil      56. 51.100 / 56. 51.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libavcodec     58. 91.100 / 58. 91.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libavformat    58. 45.100 / 58. 45.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libavdevice    58. 10.100 / 58. 10.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libavfilter     7. 85.100 /  7. 85.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libswscale      5.  7.100 /  5.  7.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libswresample   3.  7.100 /  3.  7.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   libpostproc    55.  7.100 / 55.  7.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security] [h264 @ 0x43cf980] Reinit context to 2304x1296, pix_fmt: yuv420p Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security] [h264 @ 0x43ce450] Stream #0: not enough frames to estimate rate; consider increasing probesize Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security] [h264 @ 0x43ce450] decoding for stream 0 failed Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security] pipe:: could not seek to position 0.500 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security] Input #0, h264, from 'pipe:': Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   Duration: N/A, bitrate: N/A Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]     Stream #0:0: Video: h264 (High), 1 reference frame, yuv420p(tv, bt709, progressive, left), 2304x1296, 15 tbr, 1200k tbn, 30 tbc Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security] Output #0, image2, to 'pipe:': Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   Metadata: Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]     encoder         : Lavf58.45.100 Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]     Stream #0:0: Video: h264 (High), 1 reference frame, yuv420p(tv, bt709, progressive, left), 2304x1296 (0x0), q=2-31, 15 tbr, 15 tbn, 15 tbc Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security] Stream mapping: Eufy camera] [Snapshot
[28/01/2022, 12:53:51] [homebridge-eufy-security]   Stream #0:0 -> #0:0 (copy) Eufy camera] [Snapshot
 Eufy camera] [Snapshot[homebridge-eufy-security] frame=    0 fps=0.0 q=-1.0 size=N/A time=00:00:00.00 bitrate=N/A speed=   0x    
 Eufy camera] [Snapshot[homebridge-eufy-security] frame=    0 fps=0.0 q=-1.0 size=N/A time=00:00:00.00 bitrate=N/A speed=   0x    
 Eufy camera] [Snapshot[homebridge-eufy-security] frame=    0 fps=0.0 q=-1.0 size=N/A time=00:00:00.00 bitrate=N/A speed=   0x    
[28/01/2022, 12:53:53] [homebridge-eufy-security] [AVIOContext @ 0x43de390] Statistics: 0 seeks, 3 writeouts Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security] No more output streams to write to, finishing. Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security] frame=    1 fps=0.6 q=-1.0 Lsize=N/A time=00:00:01.93 bitrate=N/A speed=1.07x     Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security] video:88kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security] Input file #0 (pipe:): Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security]   Input stream #0:0 (video): 29 packets read (149937 bytes);  Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security]   Total: 29 packets (149937 bytes) demuxed Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security] Output file #0 (pipe:): Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security]   Output stream #0:0 (video): 1 packets muxed (89885 bytes);  Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security]   Total: 1 packets (89885 bytes) muxed Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security] [AVIOContext @ 0x43d6fe0] Statistics: 250215 bytes read, 0 seeks Eufy camera] [Snapshot
[28/01/2022, 12:53:53] [homebridge-eufy-security] Eufy camera Eufy Station stopped the stream. Stopping stream.
[28/01/2022, 12:53:53] [homebridge-eufy-security] Stopping session with id: 0a69ac32-dbda-4a55-a285-0a7935098ac6
[28/01/2022, 12:53:53] [homebridge-eufy-security] No session to stop.
[28/01/2022, 12:53:53] [homebridge-eufy-security] Eufy camera Eufy Station stopped the stream. Stopping stream.
[28/01/2022, 12:53:53] [homebridge-eufy-security] Stopping session with id: b7f9fa56-ae8d-4f3a-9a83-70570742b995
[28/01/2022, 12:53:53] [homebridge-eufy-security] No session to stop.
[28/01/2022, 12:53:53] [homebridge-eufy-security] snapshot byte lenght: 89885
[28/01/2022, 12:53:56] [homebridge-eufy-security] Setting snapshotPromise to undefined.
[28/01/2022, 12:53:56] [homebridge-eufy-security] isLiveStreaming: false

**Comment by schliemann** [comment]
@njuart 

Yeah. I added a bunch of logging. Will change them to debug and only enabledetailedlogging of 2 will display verbose messages from ffmpeg.

**Comment by fullphat** [comment]
> Try running sudo lsof -i :41089
> 
> 
> 
> This should show you what is using this address.

Gives me two PIDs which come back to "hb-service" and "homebridge: homebridge-eufy-security"

**Comment by njuart** [comment]
Just FYI, installed yesterday 15.4 PB1 and it seems like interval of updating snapshots in Home app now is 1 minute instead of 10 seconds.

**Comment by fullphat** [comment]
> Just FYI, installed yesterday 15.4 PB1 and it seems like interval of updating snapshots in Home app now is 1 minute instead of 10 seconds.

My iPad is on iPadOS 15.4 Dev Beta 1 and it's still refreshing every 10 seconds

**Comment by njuart** [comment]
> > Just FYI, installed yesterday 15.4 PB1 and it seems like interval of updating snapshots in Home app now is 1 minute instead of 10 seconds.
> 
> My iPad is on iPadOS 15.4 Dev Beta 1 and it's still refreshing every 10 seconds
> 

Weird, on iPhone I clearly see it started to update every minute. I also got this crash just right now

![image](https://user-images.githubusercontent.com/21228802/151657854-c226100f-84e9-4531-af1a-24e82272887c.jpeg)

**Comment by schliemann** [comment]
@njuart Are you on the local network? When I'm not home the interval is also 1 minute. Pretty sure the interval is managed by IOS.

**Comment by njuart** [comment]
> @njuart Are you on the local network? When I'm not home the interval is also 1 minute. Pretty sure the interval is managed by IOS.

Yup, the weird part is that I'm on local network. Before installing 15.4 interval was 10 sec. Yeah, I'm pretty sure it's managed by iOS, just pointing out that on new beta I have this behavior. Other than that, and a couple of crashes, snapshots work amazing. Thx 👍

**Comment by schliemann** [comment]
@njuart I can confirm. iOS 15.3 changed the snapshot interval.

**Comment by njuart** [comment]
> @njuart I can confirm. iOS 15.3 changed the snapshot interval.

Hope it's just beta tuning.

**Comment by schliemann** [comment]
I'm not on beta. Just regular 15.3.

**Comment by schliemann** [comment]
@njuart, @fullphat, @lenoxys 

1.1.1-beta.15 is ready. Just includes snapshots and some changed settings for ffmpeg logging. "+verbose" is only set if enableDetailedLogging is 1 or above.

**Comment by njuart** [comment]
> @njuart, @fullphat, @lenoxys 
> 
> 
> 
> 1.1.1-beta.15 is ready. Just includes local live snapshots and some changed settings for ffmpeg logging. "+verbose" is only set if enableDetailedLogging is 1 or above.

Seems to be even more stable now, and log is a way much cleaner =) Thx, will keep you posted how it goes through the day

**Comment by fullphat** [comment]
> > @njuart, @fullphat, @lenoxys 
> > 
> > 
> > 
> > 1.1.1-beta.15 is ready. Just includes local live snapshots and some changed settings for ffmpeg logging. "+verbose" is only set if enableDetailedLogging is 1 or above.
> 
> Seems to be even more stable now, and log is a way much cleaner =) Thx, will keep you posted how it goes through the day

Installed as well.  All good here once I'd fixed the video codec issue.  Will continue to test as well. 👍🏼

**Comment by onkelfarmor** [comment]
Installed the new version. Snapshots working on Wireless Doorbell now - hope my battery will survive … But video stream still fails. Should I do something extra?

**Comment by njuart** [comment]
I've noticed that on beta 15 stream goes live for 3-5 seconds and then it stops with No Response.

**Comment by schliemann** [comment]
@njuart Odd. Works here. Can you share your log? See what the exception is?

**Comment by schliemann** [comment]
> Installed the new version. Snapshots working on Wireless Doorbell now - hope my battery will survive … But video stream still fails. Should I do something extra?

Please share your log.

**Comment by onkelfarmor** [comment]
> > Installed the new version. Snapshots working on Wireless Doorbell now - hope my battery will survive … But video stream still fails. Should I do something extra?
> 
> Please share your log.

[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] DEBUG: Received local livestream.
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] DEBUG: Doorbell Snapshot command: ffmpeg -probesize 3000 -analyzeduration 0 -ss 00:00:00.500 -i pipe: -frames:v 1 -c:v copy -f image2 - -hide_banner -loglevel +verbose true
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR: ffmpeg version 4.3.1 Copyright (c) 2000-2020 the FFmpeg developers Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   built with gcc 9.3.0 (Alpine 9.3.0) Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   configuration: --pkgconfigdir=/build/workspace/lib/pkgconfig --prefix=/build/workspace --pkg-config-flags=--static --extra-cflags=-I/build/workspace/include --extra-ldflags=-L/build/workspace/lib --extra-libs='-lpthread -lm' --enable-static --disable-debug --disable-shared --disable-ffplay --disable-doc --enable-openssl --enable-gpl --enable-version3 --enable-nonfree --enable-pthreads --enable-libvpx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libx264 --enable-runtime-cpudetect --enable-libfdk-aac --enable-avfilter --enable-libopencore_amrwb --enable-libopencore_amrnb --enable-filters --enable-decoder=h264 --enable-network --enable-protocol=tcp --enable-libspeex --enable-demuxer=rtsp --enable-indev=alsa --enable-outdev=alsa Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libavutil      56. 51.100 / 56. 51.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libavcodec     58. 91.100 / 58. 91.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libavformat    58. 45.100 / 58. 45.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libavdevice    58. 10.100 / 58. 10.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libavfilter     7. 85.100 /  7. 85.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libswscale      5.  7.100 /  5.  7.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libswresample   3.  7.100 /  3.  7.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR:   libpostproc    55.  7.100 / 55.  7.100 Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR: [h264 @ 0x56321db4b100] Reinit context to 1600x1200, pix_fmt: yuv420p Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR: [h264 @ 0x7f49e08c9840] Stream #0: not enough frames to estimate rate; consider increasing probesize Doorbell] [Snapshot
[1/30/2022, 10:49:38 PM] [EufySecurity-1.1.1-beta.15] ERROR: [h264 @ 0x7f49e08c9840] decoding for stream 0 failed Doorbell] [Snapshot

**Comment by schliemann** [comment]
@onkelfarmor Your log seems to indicate the snapshot is the issue? Try and increase probesize if you have access to the js file. The probesize and analyse duration are tricky since we want the fastest snapshot possible but of course without exception.

**Comment by njuart** [comment]
> @njuart Odd. Works here. Can you share your log? See what the exception is?
> 
> 
> 
> https://user-images.githubusercontent.com/10818044/151719123-cbbb04f3-fb9a-4acf-9c8c-56841b9d5f4e.MOV

I can confirm for sure, double checked couple of times, with Enable Debug flag it's not reproduced, as soon as I disable Debug, stream cuts off after 3 seconds 😵‍💫

**Comment by schliemann** [comment]
@njuart Hehe. Does the log show anything without enabledetailedlogging of 0?

**Comment by onkelfarmor** [comment]
> @onkelfarmor Your log seems to indicate the snapshot is the issue? Try and increase probesize if you have access to the js file. The probesize and analyse duration are tricky since we want the fastest snapshot possible but of course without exception.

Where do I find the js file? And what is the name?

I get the snapshots, the errors start, when I try stream

[Doorbell] The image snapshot handler for the given accessory is slow to respond! See https://git.io/JtMGR for more info.
[1/30/2022, 11:03:45 PM] [EufySecurity] Doorbell Starting video stream: 1280 x 720, 30 fps, 299 kbps (AAC-eld)
[1/30/2022, 11:03:45 PM] [EufySecurity] Doorbell_video Error in NamePipe listen EACCES: permission denied ./2.sock Error: listen EACCES: permission denied ./2.sock
    at Server.setupListenHandle [as _listen2] (node:net:1317:21)
    at listenInCluster (node:net:1382:12)
    at Server.listen (node:net:1480:5)
    at new NamePipeStream (/homebridge/node_modules/homebridge-eufy-security/src/accessories/UniversalStream.ts:54:21)
    at StreamInput (/homebridge/node_modules/homebridge-eufy-security/src/accessories/UniversalStream.ts:83:12)
    at StreamingDelegate.startStream (/homebridge/node_modules/homebridge-eufy-security/src/accessories/streamingDelegate.ts:428:41)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[1/30/2022, 11:03:45 PM] [EufySecurity] Doorbell_video FFmpeg exited with code: 1 and signal: null (Error)
[1/30/2022, 11:03:45 PM] [EufySecurity] Stopping session with id: 8f8923d2-adbc-4e18-9398-a6adc88327fb
[1/30/2022, 11:03:45 PM] [EufySecurity] No session to stop.
[1/30/2022, 11:03:51 PM] [EufySecurity] Stopping session with id: 8f8923d2-adbc-4e18-9398-a6adc88327fb
[1/30/2022, 11:03:51 PM] [EufySecurity] Doorbell Eufy Station stopped the stream. Stopping stream.
[1/30/2022, 11:03:51 PM] [EufySecurity] Stopping session with id: 8f8923d2-adbc-4e18-9398-a6adc88327fb
[1/30/2022, 11:03:51 PM] [EufySecurity] Doorbell Stopped video stream.
[1/30/2022, 11:03:51 PM] [EufySecurity] Doorbell Stopped video stream.
[1/30/2022, 11:03:51 PM] [EufySecurity] Doorbell Error occurred closing socket: Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
[1/30/2022, 11:03:51 PM] [EufySecurity] Doorbell_audio Error in NamePipe listen EACCES: permission denied ./4.sock Error: listen EACCES: permission denied ./4.sock
    at Server.setupListenHandle [as _listen2] (node:net:1317:21)
    at listenInCluster (node:net:1382:12)
    at Server.listen (node:net:1480:5)
    at new NamePipeStream (/homebridge/node_modules/homebridge-eufy-security/src/accessories/UniversalStream.ts:54:21)
    at StreamInput (/homebridge/node_modules/homebridge-eufy-security/src/accessories/UniversalStream.ts:83:12)
    at StreamingDelegate.startStream (/homebridge/node_modules/homebridge-eufy-security/src/accessories/streamingDelegate.ts:528:45)
[1/30/2022, 11:03:51 PM] [EufySecurity] Doorbell_audio FFmpeg exited with code: 1 and signal: null (Error)
[1/30/2022, 11:03:51 PM] [EufySecurity] Stopping session with id: 8f8923d2-adbc-4e18-9398-a6adc88327fb
[1/30/2022, 11:03:51 PM] [EufySecurity] No session to stop.

**Comment by schliemann** [comment]
@onkelfarmor 

Your streaming error is known. And described in this issue I believe.

https://github.com/samemory/homebridge-eufy-security/issues/308

I think we will rewrite the local streaming to not write to a local temporary file.

**Comment by njuart** [comment]
> @njuart Hehe. Does the log show anything without enabledetailedlogging of 0?

This

![image](https://user-images.githubusercontent.com/21228802/151720265-fb82f362-12e7-42a9-af42-c28a2f9b924b.png)

**Comment by onkelfarmor** [comment]
> @onkelfarmor
> 
> Your streaming error is known. And described in this issue I believe.
> 
> #308
> 
> I think we will rewrite the local streaming to not write to a local temporary file.

Sorry, didn’t get that - will try on a spare RPI instead.

**Comment by schliemann** [comment]
@njuart Might be some timing issue then. Never run without enableDetailedLogging :)

**Comment by njuart** [comment]
> @njuart Might be some timing issue then. Never run without enableDetailedLogging :)

I see, let me know if there's anything else I can help with regarding this.

**Comment by schliemann** [comment]
> > @njuart Hehe. Does the log show anything without enabledetailedlogging of 0?
> 
> This ![image](https://user-images.githubusercontent.com/21228802/151720338-4296766d-70eb-4b09-876a-acbcd94b3eee.jpeg)

Odd. Looks like the station stops the stream. Since the first line of logging originates from here:

https://github.com/samemory/homebridge-eufy-security/blob/e82d4ad866cecc09382946f8e6b7690f3557076f/src/accessories/streamingDelegate.ts#L593

Could you try the following.

1. Open home app.
2. Let snapshot generate.
3. Wait a few seconds and start the stream?

**Comment by njuart** [comment]
> https://github.com/samemory/homebridge-eufy-security/blob/e82d4ad866cecc09382946f8e6b7690f3557076f/src/accessories/streamingDelegate.ts#L593
> 
> 
> 
> Could you try the following.
> 
> 
> 
> 1. Open home app.
> 
> 2. Let snapshot generate.
> 
> 3. Wait a few seconds and start the stream?
> 
> 

Nope, still the same, stream stops after 3 seconds. So until it stops it's fine, I mean I see real stream like usual, then it stops with No Response. With Debug enabled works fine (decided to check one more time).

**Comment by schliemann** [comment]
@njuart I got home and can reproduce your error. Will take a look tonight.

**Comment by onkelfarmor** [comment]
> > @onkelfarmor
> > Your streaming error is known. And described in this issue I believe.
> > #308
> > I think we will rewrite the local streaming to not write to a local temporary file.
> 
> Sorry, didn’t get that - will try on a spare RPI instead.

Works perfect in a non Docker install.

**Comment by schliemann** [comment]
> > > @onkelfarmor
> > > Your streaming error is known. And described in this issue I believe.
> > > #308
> > > I think we will rewrite the local streaming to not write to a local temporary file.
> > 
> > 
> > Sorry, didn’t get that - will try on a spare RPI instead.
> 
> Works perfect in a non Docker install.

Great. If possible this will be refactored to not use these local files.

**Comment by schliemann** [comment]
@lenoxys What is this awaited promise used for exactly? If I remove it the stream works with enableDetailedLogging of 0.

https://github.com/samemory/homebridge-eufy-security/blob/436419a121cb9fce3c61fc354f390747e049df14/src/accessories/streamingDelegate.ts#L526

**Comment by lenoxys** [comment]
it's because we receive the sound too early both stream (video and audio) have their own timeout.

**Comment by schliemann** [comment]
@lenoxys  How did you come up with 6 seconds?

**Comment by lenoxys** [comment]
> @lenoxys  How did you come up with 6 seconds?

Hum by tweaking until audio and video match correctly :*

**Comment by njuart** [comment]
Any luck with the issue by chance?

**Comment by fullphat** [comment]
Just a note to say that my T8420 now streams without any modification so I assume Bropat has fixed the client library. 👍🏼

It also looks that the HomeKit alarm accessory works properly too, but I've not tested it fully yet.

**Comment by r33ce15** [comment]
I have snapshots and stream working but only once debug is enabled without I see same errors of no response on camera and snapshots don't seem to generate 

I am however also seeing lot of messages about the snapshot taking too long to fetch and that I should run in unbridge mode?

**Comment by lenoxys** [comment]
Yeah it's a good idea to bridge the plugin

**Comment by fullphat** [comment]
> Just a note to say that my T8420 now streams without any modification so I assume Bropat has fixed the client library. 👍🏼
> 
> It also looks that the HomeKit alarm accessory works properly too, but I've not tested it fully yet.
> 

I had to manually change Beta 19 to once again force the vcodec to be "copy" so I'm not sure if this has been fixed in the client library or not now.  FWIW it'd be handy to have an option in this plug-in's settings to manually override the vcodec setting.  A check box would be fine which, if checked, would use "copy" otherwise it would use the stream-provided codec.

**Comment by r33ce15** [comment]
> Yeah it's a good idea to bridge the plugin. at least it won't slow down only Eufy accessories. I'm thinking about providing specific bridge for each camera.

I am running it inside a child bridge which did gain some performance but still receive messages looks as though the snapshot takes while to retrieve could be down to my older pi 3b rather than 4 causing the issue
